### PR TITLE
Upgrade Guava to latest release

### PR DESCRIPTION
**Pull request by amuniz**
Upgrade guava to 30.0-jre

Pending `workflow-support` release and manual testing.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by amuniz** [comment]
```
Crashed tests:
[2021-03-04T14:31:20.437Z] io.jenkins.plugins.artifact_manager_jclouds.s3.MinioIntegrationTest
[2021-03-04T14:31:20.437Z] org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[2021-03-04T14:31:20.437Z] Command was /bin/sh -c cd /home/jenkins/workspace/rtifact-manager-s3-plugin_PR-188 && /home/jenkins/tools/hudson.model.JDK/jdk11/jdk-11.0.6+10/bin/java '-javaagent:/home/jenkins/workspace/rtifact-manager-s3-plugin_PR-188@tmp/m2repo/org/jacoco/org.jacoco.agent/0.8.6/org.jacoco.agent-0.8.6-runtime.jar=destfile=/home/jenkins/workspace/rtifact-manager-s3-plugin_PR-188/target/jacoco.exec,excludes=**/Messages.class' -Xms768M -Xmx768M -Djava.awt.headless=true -XX:+HeapDumpOnOutOfMemoryError -XX:+TieredCompilation -XX:TieredStopAtLevel=1 -jar /home/jenkins/workspace/rtifact-manager-s3-plugin_PR-188/target/surefire/surefirebooter6239230974998199472.jar /home/jenkins/workspace/rtifact-manager-s3-plugin_PR-188/target/surefire 2021-03-04T13-36-21_045-jvmRun1 surefire606850907179615346tmp surefire_013671581806891952000tmp
[2021-03-04T14:31:20.437Z] Error occurred in starting fork, check output in log
[2021-03-04T14:31:20.437Z] Process Exit Code: 143
[2021-03-04T14:31:20.437Z] Crashed tests:
[2021-03-04T14:31:20.437Z] io.jenkins.plugins.artifact_manager_jclouds.s3.MinioIntegrationTest
[2021-03-04T14:31:20.437Z] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:690)
[2021-03-04T14:31:20.437Z] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:285)
[2021-03-04T14:31:20.437Z] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:248)
```
Possibly related to https://github.com/jenkinsci/artifact-manager-s3-plugin/pull/187 ?

**Comment by amuniz** [comment]
It's passing locally BTW

**Comment by amuniz** [comment]
Green. I'm trying to do some manual tests (having issues with s3 access, not related to the plugin though).

**Comment by amuniz** [comment]
Successful local testing. Ready for review and merge.

**Comment by jglick** [comment]
> Possibly related to #187 ?

Probably.

```
[2021-03-04T13:41:14.699Z] [INFO] Running io.jenkins.plugins.artifact_manager_jclouds.s3.MinioIntegrationTest
[2021-03-04T14:31:06.617Z] Sending interrupt signal to process
```

Looks like it hanged and was killed by a stock 1h timeout. :shrug: will keep an eye on it.

File: pom.xml [hunk 1]
```diff
@@ -90,6 +94,12 @@
         <classifier>tests</classifier>
         <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci.plugins.workflow</groupId>
+      <artifactId>workflow-support</artifactId>
+      <version>3.8</version>
```
Comment (jglick) [inline pom.xml:100]: Pending a BOM update I guess?

File: pom.xml [hunk 2]
```diff
@@ -236,7 +246,7 @@
     <dependency> <!-- see maskClasses config -->
       <groupId>com.google.guava</groupId>
       <artifactId>guava</artifactId>
-      <version>20.0</version>
+      <version>30.0-jre</version>
```
Comment (jglick) [inline pom.xml:249]:
```suggestion
      <version>30.1-jre</version>
```

? https://github.com/google/guava/releases/tag/v30.1

**Review by jglick — COMMENTED** [review]
You forgot to delete https://github.com/jenkinsci/artifact-manager-s3-plugin/blob/master/.github/dependabot.yml#L11

**Comment by jglick** [comment]
Manual sanity check worked fine. Please let Dependabot run free; otherwise looks ready to release to me.

File: pom.xml (hunk 1 above)
Comment (amuniz) [reply pom.xml]: Yep, added a comment.

File: pom.xml [hunk 3]
```diff
@@ -90,6 +94,12 @@
         <classifier>tests</classifier>
         <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci.plugins.workflow</groupId>
+      <artifactId>workflow-support</artifactId>
+      <version>3.8</version> <!-- Remove when https://github.com/jenkinsci/bom/commit/4f5b55b15a72ec92ac9e59d2dc1fd9c0c4cdc7c9 is picked up -->
```
Comment (jglick) [inline pom.xml:100]: Released as `bom` version 26.

Comment (amuniz) [reply pom.xml]: Updated.

File: pom.xml [hunk 4]
```diff
@@ -90,6 +94,11 @@
         <classifier>tests</classifier>
         <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci.plugins.workflow</groupId>
+      <artifactId>workflow-support</artifactId>
+      <scope>test</scope>
+    </dependency>
```
Comment (jglick) [inline pom.xml:101]:
```suggestion
```

Dependency of other stuff, not needed.

**Review by jglick — APPROVED** [review]

**Comment by jglick** [comment]
> hanged and was killed by a stock 1h timeout

Likely JVM ergonomics acting up. Will need to play with setting an explicit heap size.

**Pull request merged**